data/query.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * External Dependencies
+ */
+import { isFunction } from 'lodash';
+
+/**
+ * WordPress Dependencies
+ */
+import { Component } from '@wordpress/element';
+
+const createQueryHigherOrderComponent = ( client ) => ( mapPropsToQuery, mapPropsToVariables = () => ( {} ) ) => ( WrappedComponent ) => {
```

Comment:
Would these more accurately be `mapQueryToProps` and `mapVariablesToProps` ?

Or am I wrong in thinking that these functions would take values from the query result and map them to props for the component?

Reply:
Actually, they are correctly names, these are useful if you want to create custom queries based on props. 

I can see things like `mapResultToProp` being useful as well. It could be a third optional callback.